product_extended_segmentation/model/template.py

```diff
@@ -100,6 +97,7 @@ def _factor(factor, product_efficiency, product_rounding):
                         wizard_obj.change_price(cr, uid, [wiz_id], context=ctx)
 
                         # NOTE: Write remaining fields, segmentation costs
+                    if context.get('update_avg_costs'):
```

Comment:
Where do you update this key in the dictionary.

Otherwise this will be false, could point me in the right direction, please

Regards